dotCMS/src/integration-test/java/com/dotcms/enterprise/HTMLDiffUtilTest.java

```diff
@@ -0,0 +1,272 @@
+        Mockito.doAnswer((Answer<Void>) invocation -> {
+            final String key = invocation.getArgumentAt(0, String.class);
+            final Object value = invocation.getArgumentAt(1, Object.class);
+            attributes.put(key, value);
+            return null;
+        }).when(request).setAttribute(Mockito.anyString(), Mockito.anyObject());
+
+        // Mock getAttribute
+        Mockito.doAnswer((Answer<Object>) invocation -> {
+            final String key = invocation.getArgumentAt(0, String.class);
+            return attributes.get(key);
+        }).when(request).getAttribute(Mockito.anyString());
+
+        final String diff = HTMLDiffUtil.htmlDiffPage(pageLive, systemUser, request, response);
+        Assert.assertNotEquals(NOTHING_CHANGED,diff);
+        final Document document = Jsoup.parse(diff);
+        final Elements removedElements = document.select("span.diff-html-removed");
+        for (final Element removedElement : removedElements) {
+           Assert.assertTrue(seekers.contains(removedElement.text()));
+        }
+
+        final Elements addedElements = document.select("span.diff-html-added");
+        for (final Element addedElement : addedElements) {
+            Assert.assertTrue(coneheads.contains(addedElement.text()));
+        }
+    }
+
+
+    /**
+     *
+     * @throws Exception
+     */
+    @Test
+    public void Test_Page_No_Changes_Expect_Nothing_Changed_Message() throws Exception {
+
+        final String pageName = "seekers-expect-No-difference-page";
+
+        final Contentlet contentlet = new ContentletDataGen(contentGenericId)
+                .languageId(defaultLang.getId())
+                .folder(folder)
+                .host(site)
+                .setProperty("title", "seekers")
+                .setProperty("body", "Skywarp, Starscream, Thundercracker")
+                .nextPersisted();
+
+        contentlet.setIndexPolicy(IndexPolicy.WAIT_FOR);
+        contentlet.setIndexPolicyDependencies(IndexPolicy.WAIT_FOR);
+        contentlet.setBoolProperty(Contentlet.IS_TEST_MODE, true);
+        contentletAPI.publish(contentlet, systemUser, false);
+
+       final HTMLPageAsset pageLive = new HTMLPageDataGen(folder, template).languageId(defaultLang.getId())
+                .pageURL(pageName)
+                .friendlyName(pageName)
+                .title(pageName).nextPersisted();
+
+        final MultiTree multiTreeV1 = new MultiTree(pageLive.getIdentifier(),
+                container.getIdentifier(), contentlet.getIdentifier(),
+                getDotParserContainerUUID(uuid), 0);
+        multiTreeAPI.saveMultiTree(multiTreeV1);
+        HTMLPageDataGen.publish(pageLive);
+
+        //Force a working copy identical to the original
+        final Contentlet workingPage = contentletAPI.checkout(pageLive.getInode(), systemUser, false);
+        contentletAPI.checkin(workingPage, systemUser, false);
+
+        HttpServletRequest request = mock(HttpServletRequest.class);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'request' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5511524)